test/test_type_promotion.py

```diff
@@ -79,15 +79,26 @@ def test_int_promotion(self, device):
         self.assertEqual(c.dtype, torch.int64)
 
     @float_double_default_dtype
-    def test_float_promotion(self, device):
-        a = torch.ones([4, 4, 4], dtype=torch.float, device=device)
-        b = torch.ones([4, 4, 4], dtype=torch.double, device=device)
-        c = a + b
-        self.assertEqual(c, b + b)
-        self.assertEqual(c.dtype, torch.double)
-        c = b + a
-        self.assertEqual(c, b + b)
-        self.assertEqual(c.dtype, torch.double)
+    def test_complex_float_promotion(self, device):
```

Comment:
does one of these tests cover the case of a dimensioned complex64 with a zero-dim double doesn't promote (returns complex64)? I'm not seeing it but could be missing it.

Reply:
oh I was thinking it might be, or that it would at least increase coverage. Looking at the implementation though I don't see that particular case.

Reply:
added the test! :D